optuna_dashboard/ts/components/CompareStudies.tsx

```diff
@@ -0,0 +1,279 @@
+import Divider from "@mui/material/Divider"
+import List from "@mui/material/List"
+import ListItem from "@mui/material/ListItem"
+import ListItemButton from "@mui/material/ListItemButton"
+import ListItemText from "@mui/material/ListItemText"
+import ListSubheader from "@mui/material/ListSubheader"
+import HomeIcon from "@mui/icons-material/Home"
+
+import { actionCreator } from "../action"
+import { studySummariesState, studyDetailsState } from "../state"
+import { AppDrawer } from "./AppDrawer"
+import { GraphEdfs } from "./GraphEdfs"
+import { GraphHistories } from "./GraphHistories"
+import { useHistory, useLocation } from "react-router-dom"
+
+const useQuery = (): URLSearchParams => {
+  const { search } = useLocation()
+  return useMemo(() => new URLSearchParams(search), [search])
+}
+
+const useSelectedStudies = (
+  studies: StudySummary[],
+  query: URLSearchParams
+): StudySummary[] => {
+  return useMemo(() => {
+    const selected = query.get("numbers")
+    if (selected === null) {
+      return []
+    }
+    const numbers = selected
+      .split(",")
+      .map((s) => parseInt(s))
+      .filter((n) => !isNaN(n))
+    return studies.filter(
+      (t) => numbers.findIndex((n) => n === t.study_id) !== -1
+    )
+  }, [studies, query])
+}
+
+const getStudyListLink = (numbers: number[]): string => {
+  const base = URL_PREFIX + "/compare-studies"
+  if (numbers.length > 0) {
+    return base + "?numbers=" + numbers.map((n) => n.toString()).join(",")
+  }
+  return base
+}
+
+export const CompareStudies: FC<{
+  toggleColorMode: () => void
+}> = ({ toggleColorMode }) => {
+  const theme = useTheme()
+  const query = useQuery()
+  const history = useHistory()
+
+  const action = actionCreator()
+  const studies = useRecoilValue<StudySummary[]>(studySummariesState)
+  const selected = useSelectedStudies(studies, query)
+
+  const studyListWidth = 200
+
+  useEffect(() => {
+    action.updateStudySummaries()
+  }, [])
+
+  const toolbar = <HomeIcon sx={{ margin: theme.spacing(0, 1) }} />
+
+  return (
+    <Box sx={{ display: "flex" }}>
+      <AppDrawer toggleColorMode={toggleColorMode} toolbar={toolbar}>
+        <Box sx={{ display: "flex", flexDirection: "row", width: "100%" }}>
+          <Box
+            sx={{
+              minWidth: studyListWidth,
+              overflow: "auto",
+              height: `calc(100vh - ${theme.spacing(8)})`,
+            }}
+          >
+            <List>
+              <ListSubheader sx={{ display: "flex", flexDirection: "row" }}>
+                <Typography sx={{ p: theme.spacing(1, 0) }}>
+                  {studies.length} Studies
+                </Typography>
+                <Box sx={{ flexGrow: 1 }} />
+              </ListSubheader>
+              <Divider />
+              {studies.map((study, i) => {
+                return (
+                  <ListItem key={study.study_id} disablePadding>
+                    <ListItemButton
```

Comment:
How about graying out the list items when `n_objectives` does not match?

![Screenshot 2023-02-07 18 16 31](https://user-images.githubusercontent.com/5564044/217202771-53d93c71-ced7-4a76-b76f-2fcf6fbd177d.png)

Reply:
As we discussed it, we will show an error message if a user tries to add a study which has different objective numbers.

Reply:
Seems to work, thanks.